src/Compilers/CSharp/Test/Emit/CodeGen/PatternTests.cs

```diff
@@ -4266,7 +4266,7 @@ .maxstack  2
 ");
             compVerifier.VerifyIL("C.M2(int?)", @"
 {
-  // Code size       23 (0x17)
+  // Code size       15 (0xf)
```

Comment:
Given the behavior of M1 versus M2 in this test, perhaps we should file an item separately for optimizing out unnecessary 'HasValue' checks in patterns.

Reply:
I opened https://github.com/dotnet/roslyn/issues/60413

Reply:
isn't it very similar to [#42912](https://github.com/dotnet/roslyn/issues/42912) ?